### DISCUSSION TITLE
Unable to receive on Ubuntu 22.04

### DISCUSSION THREAD

**Discussion by junoslukan** [localsend/localsend#573] [Discussion (English, International)] [answered]
Hi, can someone help me troubleshoot an issue I am facing on Ubuntu 22.04, please?

I can successfully send files from Ubuntu to Android, but cannot receive them in other directions. I can receive files on the same machine (with the same IP address) when running Windows, so I assume this is a configuration issue on Ubuntu.

I checked that the port specified in Localsend on both devices is 53317 and Encryption is ON on both.

The output of `netstat -atun | grep 53317` is:

```console
tcp        0      0 0.0.0.0:53317           0.0.0.0:*               LISTEN     
udp        0      0 0.0.0.0:53317           0.0.0.0:*                          
```

I also added a firewall rule as well as tried to disable ufw entirely:

```console
$ sudo ufw status verbose
Status: active
Logging: on (low)
Default: deny (incoming), allow (outgoing), disabled (routed)
New profiles: skip

To                         Action      From
--                         ------      ----
53317                      ALLOW IN    192.168.0.0/24            
```

The output of `telnet <target IP> 53317` on Android is:

```console
Unable to connect to remote host: No route to host.
```

Can someone help me with ideas on how to further troubleshoot this issue?

**Comment by phiharri** [comment]
_"No route to host"_ suggests a lower layer connectivity issue.

- Are you able to connect to other services on Ubuntu from Android? You could try `nc -lp 8000` to start a basic listening service and check if that's connectable by telnet on the client.
- `ip route get <ip>` and `arp -a <ip>` will let you validate the IP reachability between your hosts.

**Comment by juglanshub** [comment, marked as answer]
Hi
I have just installed LocalSend today on my phone & a latop running Linux Mint (derivative of Ubuntu)
and ran into exactly the same issue with sending from my Android 12 phone. (Receiving to the Phone works fine).
Temporarily disabling UFW on the laptop proved the issue was with the UFW firewall.
Adding UFW rules to allow IP addresses etc didnt seem to work.

I eventually added 'port 53317 ALLOWIN anywhere' & 'port 53317 ALLOWOUT anywhere' and this solved my problem.  
Port 53317 needs to be open in both directions as there is a handshake between devices when starting the send process.
LocalSend now working in both directions between Laptop & phone. 
As my home laptop is behind a LAN router & a WAN router running pfSense, I dont have any concerns with the UFW port 
being open on the home laptop.  

hope this helps

**Comment by junoslukan** [comment]
It turned to be a silly mistake on my part. I was sure that I was using UFW (if any firewall at all), but for some reason, it turned out I had `firewalld` enabled. Playing around with rules in UFW obviously did not help.
Once I added the corresponding rules to firewalld, I was also able to receive files.

Thank you very much, both @phiharri and @juglanshub for your help!

**Reply by aar-rafi** [reply]
damn, thanks. this just saved me from losing my hairs.
